PiwikTracker/Tracker.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+
+final public class Tracker: NSObject {
+    
+    let dispatcher: Dispatcher
+    var queue: Queue
+    let siteId: String
+    
+    required public init(siteId: String, queue: Queue, dispatcher: Dispatcher) {
+        self.siteId = siteId
+        self.queue = queue
+        self.dispatcher = dispatcher
+        super.init()
+        startDispatchTimer()
+    }
+    
+    private func queue(event: Event) {
+        queue.queue(item: event) { 
+            // Discussion: Do we want the possibility to dispatch synchronous?
+        }
+    }
+    
+    // MARK: dispatching
+    
+    private let dispatchCount = 20
+    public private(set) var isDispatching = false
+    
+    func dispatch () {
+        guard !isDispatching else { return }
+        guard queue.itemCount > 0 else {
+            startDispatchTimer()
+            return
+        }
+        isDispatching = true
+        dispatchBatch()
+    }
+    
+    private func dispatchBatch() {
+        queue.dequeue(withLimit: dispatchCount) { events in
+            self.dispatcher.send(events: events, success: {
+                dispatchBatch()
+            }, failure: { shouldContinue in
+                // Should the events be dispatched in exact order? If so, in case of a failure we cannot continue with the next items.
+                
+                // Also: Dequeueing items an re-queueing items in case of an error is error-prone. If the app crashes during dispatching all currently dispatching events would be lost. And requeueing results in wrong ordered events if we don't sort them otherwise.
```

Comment:
Also a question to transmitting events. What possible server errors are there and how should there be handled? Is "Try to transmit the Event at a later time." the answer to all errors? Or are there errors that should lead to discarding the Event on the sdk side? For example when transmitting Events that are invalid. Possibly because of they are to old.

Also: When transmitting multiple events at once. Should those events than be handled differently?

@mattab Can you possibly answer that?